proxy/InkAPI.cc

```diff
@@ -8175,471 +8177,163 @@ TSHttpTxnConfigStringGet(TSHttpTxn txnp, TSOverridableConfigKey conf, const char
+  m["proxy.config.http.redirection_enabled"] = TXN_CONFIG_PAIR(TS_CONFIG_HTTP_ENABLE_REDIRECTION);
+  m["proxy.config.http.redirect_use_orig_cache_key"] = TXN_CONFIG_PAIR(TS_CONFIG_HTTP_REDIRECT_USE_ORIG_CACHE_KEY);
+  m["proxy.config.http.request_header_max_size"] = TXN_CONFIG_PAIR(TS_CONFIG_HTTP_REQUEST_HEADER_MAX_SIZE);
+  m["proxy.config.http.response_header_max_size"] = TXN_CONFIG_PAIR(TS_CONFIG_HTTP_RESPONSE_HEADER_MAX_SIZE);
+  m["proxy.config.http.response_server_enabled"] = TXN_CONFIG_PAIR(TS_CONFIG_HTTP_RESPONSE_SERVER_ENABLED);
+  m["proxy.config.http.response_server_str"] = TXN_CONFIG_PAIR(TS_CONFIG_HTTP_RESPONSE_SERVER_STR, TS_RECORDDATATYPE_STRING);
+  m["proxy.config.http.send_http11_requests"] = TXN_CONFIG_PAIR(TS_CONFIG_HTTP_SEND_HTTP11_REQUESTS);
+  m["proxy.config.http.server_session_sharing.match"] = TXN_CONFIG_PAIR(TS_CONFIG_HTTP_SERVER_SESSION_SHARING_MATCH);
+  m["proxy.config.http.server_tcp_init_cwnd"] = TXN_CONFIG_PAIR(TS_CONFIG_HTTP_SERVER_TCP_INIT_CWND);
+  m["proxy.config.http.slow.log.threshold"] = TXN_CONFIG_PAIR(TS_CONFIG_HTTP_SLOW_LOG_THRESHOLD);
+  m["proxy.config.http.transaction_active_timeout_in"] = TXN_CONFIG_PAIR(TS_CONFIG_HTTP_TRANSACTION_ACTIVE_TIMEOUT_IN);
+  m["proxy.config.http.transaction_active_timeout_out"] = TXN_CONFIG_PAIR(TS_CONFIG_HTTP_TRANSACTION_ACTIVE_TIMEOUT_OUT);
+  m["proxy.config.http.transaction_no_activity_timeout_in"] = TXN_CONFIG_PAIR(TS_CONFIG_HTTP_TRANSACTION_NO_ACTIVITY_TIMEOUT_IN);
+  m["proxy.config.http.transaction_no_activity_timeout_out"] = TXN_CONFIG_PAIR(TS_CONFIG_HTTP_TRANSACTION_NO_ACTIVITY_TIMEOUT_OUT);
+  m["proxy.config.http.uncacheable_requests_bypass_parent"] = TXN_CONFIG_PAIR(TS_CONFIG_HTTP_UNCACHEABLE_REQUESTS_BYPASS_PARENT);
+  m["proxy.config.net.sock_option_flag_out"] = TXN_CONFIG_PAIR(TS_CONFIG_NET_SOCK_OPTION_FLAG_OUT);
+  m["proxy.config.net.sock_packet_mark_out"] = TXN_CONFIG_PAIR(TS_CONFIG_NET_SOCK_PACKET_MARK_OUT);
+  m["proxy.config.net.sock_packet_tos_out"] = TXN_CONFIG_PAIR(TS_CONFIG_NET_SOCK_PACKET_TOS_OUT);
+  m["proxy.config.net.sock_recv_buffer_size_out"] = TXN_CONFIG_PAIR(TS_CONFIG_NET_SOCK_RECV_BUFFER_SIZE_OUT);
+  m["proxy.config.net.sock_send_buffer_size_out"] = TXN_CONFIG_PAIR(TS_CONFIG_NET_SOCK_SEND_BUFFER_SIZE_OUT);
+  m["proxy.config.ssl.hsts_include_subdomains"] = TXN_CONFIG_PAIR(TS_CONFIG_SSL_HSTS_INCLUDE_SUBDOMAINS);
+  m["proxy.config.ssl.hsts_max_age"] = TXN_CONFIG_PAIR(TS_CONFIG_SSL_HSTS_MAX_AGE);
+  m["proxy.config.url_remap.pristine_host_hdr"] = TXN_CONFIG_PAIR(TS_CONFIG_URL_REMAP_PRISTINE_HOST_HDR);
+  m["proxy.config.websocket.active_timeout"] = TXN_CONFIG_PAIR(TS_CONFIG_WEBSOCKET_ACTIVE_TIMEOUT);
+  m["proxy.config.websocket.no_activity_timeout"] = TXN_CONFIG_PAIR(TS_CONFIG_WEBSOCKET_NO_ACTIVITY_TIMEOUT);
+
+  // m[""] = TXN_CONFIG_PAIR();
+
+  return m;
+};
+
+std::unordered_map<std::string, TXN_CONFIG_PAIR> TXN_CONFIG_LOOKUP_TABLE = create_map();
+
+TXN_CONFIG_PAIR *
+TXN_CONFIG_LOOKUP(std::string name)
```

Comment:
If you want to keep this, it should be static. I think that this is only used in one place and small enough just to inline it into ``TSHttpTxnConfigFind``.